packages/interweave-emoji-picker/src/EmojiList.js

```diff
@@ -166,56 +169,72 @@ export default class EmojiList extends React.PureComponent<EmojiListProps, Emoji
     e.stopPropagation();
     e.persist();
 
-    this.handleScrollDebounced(e.currentTarget);
+    this.handleScrollDebounced(e);
+    this.props.onScroll(e);
   };
 
   /**
    * A scroll handler that is debounced for performance.
    */
-  handleScrollDebounced = debounce((container) => {
-    this.loadEmojiImages(container);
+  handleScrollDebounced = debounce((e: SyntheticWheelEvent<HTMLDivElement>) => {
+    // $FlowIgnore
+    this.loadEmojiImages(e.target, e);
   }, SCROLL_DEBOUNCE);
 
   /**
    * Loop over each group section within the scrollable container
    * and determine the active group and whether to load emoji images.
    */
-  loadEmojiImages(container: HTMLDivElement) {
+  loadEmojiImages(container: HTMLDivElement, e?: SyntheticWheelEvent<HTMLDivElement>) {
+    const { scrollTop } = container;
     const { searchQuery } = this.props;
     const { loadedGroups } = this.state;
     let updateState = false;
+    let lastGroup = '';
 
-    Array.from(container.children).forEach((section) => {
+    Array.from(container.children).some((section) => {
```

Comment:
some?

Reply:
Because we don't need to loop through the entire list.